mhr_api/src/mhr_api/services/gcp_auth/token_service.py

```diff
@@ -0,0 +1,79 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This maintains access tokens for API calls."""
+import os
+from abc import ABC, abstractmethod
+
+import google.auth.transport.requests
+from google.oauth2 import service_account
+from flask import current_app
+
+
+class TokenService(ABC):  # pylint: disable=too-few-public-methods
```

Comment:
Do you want this to be explicit, or pick up the service account assigned to this. We're not consistent here and should choose a singular approach.

Reply:
Refactored to follow your approach on obtaining service account credentials either as one env var or from the google env.